internal/common/utils.go

```diff
@@ -33,6 +67,16 @@ func CreateTestRepository() (string, error) {
 	if err != nil {
 		return testDir, err
 	}
+
 	cmd := exec.Command("git", "init", testDir)
-	return testDir, cmd.Run()
+	if err := cmd.Run(); err != nil {
+		return testDir, err
+	}
+
+	return testDir, InitializeGittufNamespace(testDir)
+}
+
+func GetGitDir() string {
```

Comment:
Any reason why this should be a func over a const?

(my guess based on the TODO comment is you have future plans for this)

Reply:
Git repos can be initialized with a different `$GIT_DIR` or `--separate-git-dir` (https://git-scm.com/docs/git-init#Documentation/git-init.txt---separate-git-dirltgit-dirgt). In that case, rather than a directory, we have a `.git` file:

```
~/tmp/gitobj $ cat .git
gitdir: /home/saky/tmp/gitdir
```

Eventually, this func should support parsing `.git` if it's a file to identify the actual git dir. Note, IIRC gitsign doesn't support this yet and I might submit a patch when I get the time to implement it here as well.